### ISSUE TITLE
Condition has length > 1 error when removing chimeras and/or assigning taxonomy

### ISSUE THREAD

**Issue by weeds23** [benjjneb/dada2#1633]
Hi Ben,
I am encountering problems with the assignTaxonomy (and other) functions in dada2 v.1.24.0. I have been following the tutorial and get all the way through to the assignTaxonomy step, but then it just runs and runs for days. So, I used your suggested fix to revert back to an earlier version of dada2:
>devtools::install_github("benjjneb/dada2",ref="taxonomy-segfault-fix")

This installs v.1.15.1. However, then I encounter the following error:
>taxa <- assignTaxonomy(seqtab.nochim, unite.ref, multithread = TRUE, tryRC = TRUE)
Error in if (class(object) == "DNAStringSet") { : the condition has length > 1

After the reversion, this error then also appears further up the pipeline anywhere after the creation of the sequence table. For example,

>table(nchar(getSequences(seqtab))) 
Error in h(simpleError(msg, call)) : 
error in evaluating the argument 'x' in selecting a method for function 'nchar': the condition has length > 1

> seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
Error in if (class(unqs) != "list") { : the condition has length > 1

If I understand correctly, this error implies that the sequence table is not in the expected format. However, I have checked that everything looks as it should be, is the correct class, that the sequence table is a proper matrix. And I have compared my sequence table to the example one here: https://www.rdocumentation.org/packages/dada2/versions/1.0.3/topics/makeSequenceTable and everything looks as it should be. Also, when I also run any of the above functions on the example sequence table specified in that link, it also gives me the same error: 

>derep1 <- derepFastq(system.file("extdata", "sam1F.fastq.gz", package="dada2"))
>derep2 <- derepFastq(system.file("extdata", "sam2F.fastq.gz", package="dada2"))
>dada1 <- dada(derep1, tperr1)
>dada2 <- dada(derep2, tperr1)
>seqt = makeSequenceTable(list(sample1=dada1, sample2=dada2))
>seqt.nochim <- removeBimeraDenovo(seqt, method="consensus", multithread=TRUE, verbose=TRUE)
Error in if (class(unqs) != "list") { : the condition has length > 1

Any suggestions as to what this error is referring to, and why it only appears in v.1.15.1?
Sarah

**Comment by benjjneb** [comment]
> So, I used your suggested fix to revert back to an earlier version of dada2

Can you clarify where I suggested this previously?

**Comment by weeds23** [comment]
Sure, here https://github.com/benjjneb/dada2/issues/916#issuecomment-588367932.

**Comment by benjjneb** [comment]
If everything is working up until `assignTaxonomy` the most likely explanation is that you are running into memory issues at that step, which is causing it to take far too long to run. One thing to test is to use a smaller reference database (e.g. RDP), and see if that solves the `assignTaxonomy` run-time issue. Also, if you can identify the amount of RAM you have available, and perhaps identify another computing environment where you can have at least 16GB of RAM available, or even 32+.

**Comment by weeds23** [comment]
Hi Ben, and thank you for your reply. I think we came to the same conclusion. I inherited this code from a student who found the original "fix". She couldn't find the original post where she found this I'm sorry, but on looking over the ones I have found since, I don't think this was appropriate after all. Still not sure why even the examples were producing this error when using removeBimeraDenovo, but it was only with v1.15.1 anyway. And yes, I think the issue may have been RAM because I ran assigntaxonomy with a subset of the samples and it worked. Thanks again for your help and quick reply.
